### PR TITLE
OneHotEncodingTransformer support for lists and lists of lists

### PR DESCRIPTION
**Pull request by fealho**
Resolves sdv-dev/CTGAN#87

### CODE REVIEW

File: rdt/transformers/categorical.py [hunk 1]
```diff
@@ -223,28 +223,54 @@ class OneHotEncodingTransformer(BaseTransformer):
...
+        Args:
+            data (pandas.Series, numpy.ndarray, list or list of lists):
+                Data to prepare.
+
+        Returns:
+            pandas.Series or numpy.ndarray
+        """
+        if isinstance(data, list):
+            data = np.array(data)
+            if len(data.shape) == 2:
```
Comment (csala) [inline rdt/transformers/categorical.py:242]: Maybe we should put this block one level higher, outside of the `isinstance(data, list)` block?

This way we would also support 2D numpy arrays as input.

File: rdt/transformers/categorical.py [hunk 2]
```diff
@@ -223,28 +223,54 @@ class OneHotEncodingTransformer(BaseTransformer):
...
+                if data.shape[1] != 1:
+                    raise ValueError("Unexpected format.")
+                data = data[:, 0]
```
Comment (csala) [inline rdt/transformers/categorical.py:245]: Can we add a couple of blank lines to separate the blocks after unindenting?

```python
if len(data.shape) == 2:
    if data.shape[1] != 1:
        raise ValueError("Unexpected format.")

    data = data[:, 0]

elif len(data.shape) > 2:
    raise ValueError("Unexpected format.")

return data
```

**Review by csala — CHANGES_REQUESTED** [review]
Looks good! Just a couple of comments.

File: rdt/transformers/categorical.py (hunk 1 above)
Comment (fealho) [reply rdt/transformers/categorical.py:242]: Sorry, I'm a little confused here, you mean just delete the isinstance?

File: rdt/transformers/categorical.py (hunk 2 above)
Comment (fealho) [reply rdt/transformers/categorical.py:245]: Yes, this does look much more clean.

File: rdt/transformers/categorical.py (hunk 1 above)
Comment (csala) [reply rdt/transformers/categorical.py:242]: No, I mean removing the indentation of the `if len(data.shape)` lines one level, so they are always run independently on whether the input was a list or a numpy array.

This way, if the input was already a numpy array, its shape is validated too.

Comment (fealho) [reply rdt/transformers/categorical.py:242]: Oh, mean this?

```python3
if isinstance(data, list):
    data = np.array(data)

if isinstance(data, np.ndarray):
    if len(data.shape) == 2:
        if data.shape[1] != 1:
            raise ValueError("Unexpected format.")
    
        data = data[:, 0]

    elif len(data.shape) > 2:
        raise ValueError("Unexpected format.")
    
return data
```

Maybe at this point we might as well just support anything np.ndarray does? Like this:

```python3
data = np.array(data)
if len(data.shape) == 2:
    if data.shape[1] != 1:
        raise ValueError("Unexpected format.")

    data = data[:, 0]

elif len(data.shape) > 2:
    raise ValueError("Unexpected format.")

return data
```

File: rdt/transformers/categorical.py [hunk 3]
```diff
@@ -236,15 +236,21 @@ def _prepare_data(data):
 
         Returns:
             pandas.Series or numpy.ndarray
+
+                if isinstance(data, list):
```
Comment (csala) [inline rdt/transformers/categorical.py:240]: I suspect these lines have made their way here by error :-)

Comment (fealho) [reply rdt/transformers/categorical.py:240]: Oof, my bad.

**Review by csala — APPROVED** [review]

**Pull request merged**